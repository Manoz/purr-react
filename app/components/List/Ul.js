import styled from 'styled-components';

const Ul = styled.ul`
  margin: 1em 0 1em 1.11111em;
  padding: 0;

  & ul,
  & ol {
    margin-top: 0;
    margin-bottom: 0;
  }
`;

export default Ul;
