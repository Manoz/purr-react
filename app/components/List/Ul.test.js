import React from 'react';
import { shallow } from 'enzyme';

import Ul from './Ul';

describe('<Ul />', () => {
  it('should render an <ul> tag', () => {
    const wrapper = shallow(<Ul />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const wrapper = shallow(<Ul id={id} />);
    expect(wrapper.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const wrapper = shallow(<Ul attribute="test" />);
    expect(wrapper.prop('attribute')).toEqual('test');
  });
});
