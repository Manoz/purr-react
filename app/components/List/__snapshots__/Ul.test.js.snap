// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ul /> should render an <ul> tag 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Ul-sc-10uf1zs-0",
        "isStatic": true,
        "rules": Array [
          "margin:1em 0 1em 1.11111em;padding:0;& ul,& ol{margin-top:0;margin-bottom:0;}",
        ],
      },
      "displayName": "Ul",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Ul-sc-10uf1zs-0",
      "target": "ul",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
